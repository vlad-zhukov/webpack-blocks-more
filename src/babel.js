/*
 * Babel loader.
 */
export default function babel(options = {}) {
    return (context, {addLoader}) =>
        addLoader({
            ...context.match,
            use: {
                loader: 'babel-loader',
                options,
            },
        });
}
