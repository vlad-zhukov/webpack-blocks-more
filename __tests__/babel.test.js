import {createConfig, match} from '@webpack-blocks/webpack';
import babel from '../src/babel';

it('should work with match()', () => {
    const config = createConfig([match('*.js', [babel()])]);

    expect(config).toMatchSnapshot();
});

it('should take options', () => {
    const config = createConfig([
        match('*.js', [
            babel({
                compact: false,
            }),
        ]),
    ]);

    expect(config).toMatchSnapshot();
});
