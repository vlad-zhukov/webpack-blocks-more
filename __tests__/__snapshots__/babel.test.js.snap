// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take options 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.js\\$/,
        "use": Object {
          "loader": "babel-loader",
          "options": Object {
            "compact": false,
          },
        },
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with match() 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.js\\$/,
        "use": Object {
          "loader": "babel-loader",
          "options": Object {},
        },
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;
